pennylane/transforms/qcut.py

```diff
@@ -675,3 +676,163 @@ def contract_tensors(
     kwargs = {} if use_opt_einsum else {"like": tensors[0]}
 
     return contract(eqn, *tensors, **kwargs)
+
```

Comment:
Just a general remark about these functions: I suspect there could be a more efficient way to work with the execution results that obviates some of the reshaping into flat lists, but that might require looking into the code from previous PRs...

Reply:
Nice, yes would be good to understand that more! Perhaps once the full functionality has been added, we can do a performance benchmark to see where the bottlenecks are?